external/pdal_wrench/info.cpp

```diff
@@ -0,0 +1,185 @@
+/*****************************************************************************
+ *   Copyright (c) 2023, Lutra Consulting Ltd. and Hobu, Inc.                *
+ *                                                                           *
+ *   All rights reserved.                                                    *
+ *                                                                           *
+ *   This program is free software; you can redistribute it and/or modify    *
+ *   it under the terms of the GNU General Public License as published by    *
+ *   the Free Software Foundation; either version 3 of the License, or       *
+ *   (at your option) any later version.                                     *
+ *                                                                           *
+ ****************************************************************************/
+
+#include <iostream>
+#include <filesystem>
+#include <thread>
+
+#include <pdal/PipelineManager.hpp>
+#include <pdal/Stage.hpp>
+#include <pdal/util/ProgramArgs.hpp>
+#include <pdal/pdal_types.hpp>
+#include <pdal/Polygon.hpp>
+
+#include <gdal/gdal_utils.h>
+
+#include "utils.hpp"
+#include "alg.hpp"
+#include "vpc.hpp"
+
+using namespace pdal;
+
+namespace fs = std::filesystem;
+
+
+void Info::addArgs()
+{
+}
+
+bool Info::checkArgs()
+{
+    return true;
+}
+
+
+// extract information from WKT - write to "crs" and "units" strings
+static void formatCrsInfo(const std::string &crsWkt, std::string &crs, std::string &units)
+{
+  if (crsWkt.empty())
+  {
+      crs = "(unknown)";
+      units = "(unknown)";
+  }
+  else if (crsWkt == "_mix_")
+  {
+      // this can only happen in case of VPC - it is our special marker
+      // when there are two or more different CRS in the input files
+      crs = "(multiple)";
+      units = "(unknown)";
+  }
+  else
+  {
+      // this code is quite clumsy... most of what we need is handled by pdal::SpatialReference, but:
+      // - there's no API to get human readable name of the CRS
+      // - https://github.com/PDAL/PDAL/issues/3943 - SpatialReference::identifyVerticalEPSG() is broken
+      // - https://github.com/PDAL/PDAL/issues/3946 - SpatialReference::getHorizontalUnits() may return incorrect units
+      pdal::SpatialReference sr(crsWkt);
+      std::string wktHoriz = sr.getHorizontal();
+      std::string wktVert = sr.getVertical();
+      CRS c(crsWkt);
+      CRS ch(wktHoriz);
+      CRS cv(wktVert);
+      std::string crsEpsg = c.identifyEPSG();
```

Comment:
what happens with non-epsg crses here?

Reply:
In that case the info tool only prints the CRS name from WKT, without EPSG code in parentheses